src/modules/navigation/duck/hooks.ts

```diff
@@ -1,33 +1,93 @@
-import { useCallback } from 'react'
+import { useCallback, useRef } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 
+import { useClient, Q, useQuery } from 'cozy-client'
 import flag from 'cozy-flags'
 
 import { DEFAULT_SORT, SORT_BY_UPDATE_DATE } from '@/config/sort'
 import { TRASH_DIR_ID } from '@/constants/config'
+import { DOCTYPE_DRIVES_SETTINGS } from '@/lib/doctypes'
+import logger from '@/lib/logger'
 import { sortFolder, getSort } from '@/modules/navigation/duck'
 
+export interface Sort {
+  attribute: string
+  order: string
+}
+
+interface QueryResult {
+  data: [
+    | {
+        attributes: Sort
+      }
+    | undefined
+  ]
+  fetchStatus?: string
+}
+
 const useFolderSort = (folderId: string): [Sort, (props: Sort) => void] => {
+  const client = useClient()
+
   const defaultSort: Sort =
-    flag('drive.default-updated-at-sort.enabled') || folderId === TRASH_DIR_ID
+    flag('drive.default-updated-at-sort.enabled') ||
+    folderId === TRASH_DIR_ID ||
+    folderId === 'recent'
       ? SORT_BY_UPDATE_DATE
       : DEFAULT_SORT
+
   const dispatch = useDispatch()
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  const currentSort = (useSelector(state => getSort(state)) ||
-    defaultSort) as Sort
-  const setOrder = useCallback(
-    ({ attribute, order }: Sort) => {
+
+  const driveSettingsResult = useQuery(Q(DOCTYPE_DRIVES_SETTINGS), {
+    as: DOCTYPE_DRIVES_SETTINGS,
+    enabled: !!client
```

Comment:
What is the purpose of the `enabled: !!client`? I never saw it in Cozy codebase (but maybe there is a good reason)